Reorganise and clean up the whole guide.

m-ou-se opened the pull request:
This reorganises the guide and removes a lot of stub sections.

jyn514 commented on src/feature-lifecycle/tracking-issues.md:
```diff
@@ -1,16 +0,0 @@
-
-## Creating a tracking issue
-
-There's a template that can be used to fill out the initial tracking issue. The Libs team also maintains [a Cargo tool](https://github.com/rust-lang/libs-team/tree/main/tools/unstable-api) that can be used to quickly dump the public API of an unstable feature.
```
can you please link the `unstable-api` tool from the libs tracking issue template? https://github.com/rust-lang/rust/issues/new?template=library_tracking_issue.md

m-ou-se replied:
I'm pretty sure the tool is broken. It hasn't worked for a long time now. That's why I removed it.

jyn514 replied:
😞 fair enough

jyn514 reviewed (commented):
this gets rid of a lot of the "broken windows", thanks :) reading the diff it does strike me that the "for reviewers" sections were acting as little summaries where you only have to read the bullet points rather than the whole section, but maybe people weren't using them? i don't feel strongly about keeping them around.

m-ou-se commented on src/feature-lifecycle/api-change-proposals.md:
It's outdated and will be replaced by a new policy. See https://github.com/rust-lang/std-dev-guide/pull/55

m-ou-se commented on src/SUMMARY.md:
```diff
@@ -1,53 +1,30 @@
-[Getting started](./getting-started.md)
+- [About this Guide](about.md)
+
+- [Contributing to the standard libraries]()
```
Yes, that's better than having a link that just goes to an empty filler page.

m-ou-se commented on src/tools-and-bots/bors.md:
````diff
@@ -1,15 +0,0 @@
-@bors r+
-```
-
-## Rolling up
-
-For Libs PRs, rolling up is usually fine, in particular if it's only a new unstable addition or if it only touches docs. See the [rollup guidelines](https://forge.rust-lang.org/compiler/reviews.html#rollups) for more details on when to rollup.
````
Added it back to team/reviewing.

m-ou-se commented on src/feature-lifecycle/deprecation.md:
There was almost nothing on this page. It says that APIs can't be removed from stable, but that doesn't need a separate page next to the 'breaking changes' page. The remaining info it had on *how* to handle a deprecation is outdated/incomplete. Once we properly figure out when and how to deprecate things, we should add that as a page under the new `policies/` section.

m-ou-se commented:
@rust-lang/libs This reorganises and cleans up the std-dev-guide a bit. We had a lot of nearly empty pages and a structure that made it somewhat hard to just incrementally add new information. I removed all empty stubs and not-so-useful information and tried to categorize what remained into simpler categories. ("how to contribute", "policies", "team", etc.)

See https://std-dev-guide.rust-lang.org/

I'm not particularly tied to this structure/categorization; it's just what seemed to work well for the information we currently have. Feel free to move everything around if you think another structure works better. :)

I might have accidentally also removed some things I miscategorized as unuseful, as I tried to err a bit on the side of cleaning up too much rather than cleaning up too little, considering the amount of incomplete and outdated info there was. ^^
